src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,89 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    private static final Map<Class<?>, Class<?>> implementations = new HashMap<>();
+    private static final Map<Class<?>, Object> components = new HashMap<>();
     private static final Injector injector = new Injector();
 
+    private Injector() {
+        fillMap();
+    }
+
+    private static void fillMap() {
+        implementations.put(FileReaderService.class,
+                FileReaderServiceImpl.class);
+        implementations.put(ProductParser.class,
+                ProductParserImpl.class);
+        implementations.put(ProductService.class,
+                ProductServiceImpl.class);
+    }
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implementation = getImpl(interfaceClazz);
+        Object clazzImplementationsInstance = instanceOf(implementation);
+        Field[] fields = implementation.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object value = getInstance(field.getType());
+                field.setAccessible(true);
```

Comment:
put this line inside the try block, cause `setAccessible()` would throw an exception too. [See documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible(boolean))